### PR TITLE
Fix to the header select all checkbox client side pagination bug and Configurable footer postion

### PR DESCRIPTION
**Pull request by madhuinwiter**
In client side pagination select all checkbox selecting all the records(rows) in all pages. This fix is to select only existing records(rows) in the current page.

**What kind of change does this PR introduce?** (check one with "x")
- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
There is on PR(#736) for this. But have lot of conflicts and not fixed it completly.

_**Client side pagination:**_
If I click select all check box in the header while I have pagination it returning all the rows of all pages. 

**_New feature: Footer Position_** 
There is one PR #751 for this.
Footer(Pagination with summary) display at the end of the table only.

**What is the new behavior?**

_**Client side pagination:**_
If I click select all check box in the header while I have pagination it will return only rows of the current page.
I am going to remove selected rows in the following usecases
1. On page change
2. On coulmn sorting

**_New feature: Footer Position_**
PR #751 will be solved.
Footer(Pagination with summary) display is now configurable with the following options for **footerPosition** input varaible
1. Bottom 2. Top 3. Both

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: src/components/datatable.component.ts [hunk 1]
```diff
@@ -848,6 +842,12 @@ export class DatatableComponent implements OnInit, AfterViewInit, DoCheck {
    * @memberOf DatatableComponent
    */
   onBodyPage({offset}: any): void {
+    this.selected = [];
+    this.allRowsSelected = false;
+    this.select.emit({
```
Comment (amcdnl) [inline src/components/datatable.component.ts]: Why would it do this on every pagination?

File: src/components/datatable.component.ts [hunk 2]
```diff
@@ -23,6 +23,22 @@ import { MouseEvent } from '../events';
     <div
       visibilityObserver
       (visible)="recalculate()">
+      <datatable-footer
```
Comment (amcdnl) [inline src/components/datatable.component.ts]: I like this, can you do this in a separate PR?

File: src/components/datatable.component.ts (hunk 1 above)
Comment (madhuinwiter) [inline src/components/datatable.component.ts]: Beacuse we need to clear the selected rows on page change or sorting. The main goal of this is to select rows on the current page only.
Thanks.

File: src/components/datatable.component.ts (hunk 2 above)
Comment (madhuinwiter) [inline src/components/datatable.component.ts]: @amcdnl Sure will do.

**Comment by amcdnl** [comment]
Side note: I manually merged ur footer fix in 10.0.0

File: src/components/datatable.component.ts (hunk 2 above)
Comment (madhuinwiter) [inline src/components/datatable.component.ts]: @amcdnl riased new PR #908  fro this.
